### PR TITLE
feat: add TCP connection state sampler

### PR DESCRIPTION
**Pull request by brayniac**
Tracks TCP connections by state.

### CODE REVIEW

File: src/samplers/tcp/stats.rs [hunk 1]
```diff
@@ -30,6 +35,114 @@ counter_with_histogram!(
     "tcp/transmit/retransmit",
     "number of TCP segments retransmitted"
 );
+counter_with_histogram!(
+    TCP_CONN_ACTIVE_OPEN,
+    TCP_CONN_ACTIVE_OPEN_HISTOGRAM,
+    "tcp/connection/active_open",
+    "number of TCP active opens"
+);
+counter_with_histogram!(
```
Comment (mihirn) [inline src/samplers/tcp/stats.rs:44]: Where are these two histograms being used?

Comment (brayniac) [reply src/samplers/tcp/stats.rs:44]: Whoops. Wasn't supposed to include those in this PR.

File: src/samplers/tcp/connection_state/mod.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                    return;
+                }
+
+                for line in data.lines() {
+                    let parts: Vec<&str> = line.split_whitespace().collect();
+
+                    // find and increment the temporary gauge for this state
+                    if let Some(Ok(state)) = parts.get(3).map(|v| u8::from_str_radix(v, 16)) {
+                        if let Some((_, gauge)) = self.gauges.get_mut(state as usize - 1) {
+                            *gauge += 1;
```
Comment (mihirn) [inline src/samplers/tcp/connection_state/mod.rs:107]: Not familiar with the data format so just checking: are we expecting to increment the gauge by only 1 every time as opposed to some value that is ready from the file?

**Review by mihirn — APPROVED** [review]

File: src/samplers/tcp/connection_state/mod.rs (hunk 2 above)
Comment (brayniac) [reply src/samplers/tcp/connection_state/mod.rs:107]: Yes, each open TCP socket is represented by a single line in the file. So we are grabbing the state field (4th column) and incrementing the corresponding state for each socket.

Comment (mihirn) [reply src/samplers/tcp/connection_state/mod.rs:107]: Ack, thanks!

**Pull request merged**